app/routes/api.js

```diff
@@ -383,16 +387,79 @@ apiRouter.post("/accounts/add", (req, res) => {
         return res.sendStatus(400);
     }
 
-    const { username, firstname, lastname, accountType } = req.body;
+    const { username, password, firstname, lastname} = req.body;
+
+    await pool.query("SELECT * FROM accounts WHERE username = $1", [username])
+    .then(async result => {
+        if (result.rows.length > 0) {
+            res.statusMessage = "Username Already Exists";
+
+            return res.sendStatus(400);
+        }
+
+        let hash = "";
+    
+        hash = await argon2.hash(password);
+
+        query("INSERT INTO accounts (username, password, first_name, last_name, account_type) VALUES ($1, $2, $3, $4, $5)",
+            [username, hash, firstname, lastname, "admin"],
+            res,
+            true,
+        );
+    })
+    
+});
+
+apiRouter.put("/auth/pos/account/modify/:id", async (req, res) => {
+    const body = req.body;
+   
+    const id = req.params.id;
+
+    const { username, password, firstname, lastname} = body;
+
+    const result = await pool
+        .query("SELECT * FROM accounts WHERE username = $1", [body.username])
+        .then((result) => {
+            if (result.rows.length === 0) return res.sendStatus(400);
+
+            return result.rows[0];
+        })
+        .catch((error) => {
+            console.log(error);
+            return res.sendStatus(500);
+        });
+
+    const { dbPassword } = result;
+
+    const verified = await argon2.verify(dbPassword, password);
```

Comment:
```suggestion
    const dbPassword = await pool
        .query("SELECT * FROM accounts WHERE username = $1", [body.username])
        .then((result) => {
            if (result.rows.length === 0) return res.sendStatus(400);

            return result.rows[0].password;
        })
        .catch((error) => {
            console.log(error);
            return res.sendStatus(500);
        });

    const verified = await argon2.verify(dbPassword, password);
```

Reply:
Made changes before I saw this. The new code is much much worse so if this is optimal lmk